
This patch removes avctx context reinit in h264_decode_flush function.

Reinitialization avctx context triggers ff_v4l2_request_uninit and then
ff_v4l2_request_init calls and results with lost reference between ffmpeg
buffers and mythtv video buffers (which have their own flush at seek method.

This makes ffmpeg h264 context is not reinitialized in the same way like no
context reinicjalization on vp8/vp9 & hevc codecs.


diff --speed-large-files --no-dereference --minimal -Naur mythtv-fba18601ff296e3becf4e9d674930b4645497cf3/mythtv/external/FFmpeg/libavcodec/h264dec.c mythtv-fba18601ff296e3becf4e9d674930b4645497cf3/mythtv/external/FFmpeg/libavcodec/h264dec.c
--- mythtv-fba18601ff296e3becf4e9d674930b4645497cf3/mythtv/external/FFmpeg/libavcodec/h264dec.c	2021-12-20 11:55:24.053182458 +0100
+++ mythtv-fba18601ff296e3becf4e9d674930b4645497cf3/mythtv/external/FFmpeg/libavcodec/h264dec.c	2021-12-20 11:54:48.693183096 +0100
@@ -477,9 +477,6 @@
     ff_h264_unref_picture(h, &h->cur_pic);
 
     h->mb_y = 0;
-
-    ff_h264_free_tables(h);
-    h->context_initialized = 0;
 }
 
 static int get_last_needed_nal(H264Context *h)
